spec/system/topic_list/glimmer_spec.rb

```diff
@@ -20,6 +20,23 @@
     end
   end
 
+  describe "/new" do
+    let(:topic_list) { PageObjects::Components::TopicList.new }
+
+    it "shows the list and the toggle buttons" do
+      SiteSetting.experimental_new_new_view_groups = group.name
+      Fabricate(:topic)
+      Fabricate(:new_reply_topic, current_user: user)
```

Comment:
We could extract this to a `before` block just above, this way we don’t “pollute” the example with setup code. Wdyt? :slightly_smiling_face:

Reply:
🙅‍♂️😄

I find "Arrange Act Assert" the most useful and readable. Extracting the first part to a separate block doesn't really bring any benefits 😅 And splitting it into three separate ones like:

```rb
before do
  # arrange
end

subject(:a_required_name) { act }

it "shows whatever" do
  # assert
end
```

…would probably also require wrapping in a `context` block, if you wanted to add more tests in the same `describe` block.
We doubled the line and character count while the things still work the same 😅 

Want to plop a `binding.pry` in there and step through the spec? Then you bounce through rspec internals 🥲 

Want to copy and paste some code to the rails console to try something out? You can't select a nice chunk of code but a bit of this block, a bit of this line, etc.

---

I accept things like `fab!()` because those are important for the performance of our specs.
And using `let` and `before` for stuff that is required by every single spec in a file (if there are many of them, if not - inline, inline, inline 😄)

Reply:
Yes, on adding lines, but one of the ideas behind RSpec is the living documentation part. It works rather well when you follow the established conventions (and you’re used to it, I guess :sweat_smile:). So you define a context, right after it, you describe the related setup, and you move on to examples (or more contexts, usually they reflect how many conditionals there are in the code, which can help to see bad designed code for example). 
I don’t remember a time when it prevented me from debugging things either :thinking: 

But no worries, I don’t want to fight over this, we’ll probably have the opportunity to talk about it in another context :grin: